src/unix/dlunix.cpp

```diff
@@ -106,22 +106,21 @@ wxDllType wxDynamicLibrary::RawLoad(const wxString& libname, int flags)
 }
 
 /* static */
-void wxDynamicLibrary::Unload(wxDllType handle)
+bool wxDynamicLibrary::Unload(wxDllType handle)
 {
-#ifdef HAVE_DLERROR
     int rc =
-#endif
 
 #ifdef USE_POSIX_DL_FUNCS
     dlclose(handle);
 #else // !USE_POSIX_DL_FUNCS
     shl_unload(handle);
 #endif // USE_POSIX_DL_FUNCS/!USE_POSIX_DL_FUNCS
 
-#if defined(USE_POSIX_DL_FUNCS) && defined(HAVE_DLERROR)
     if ( rc != 0 )
-        Error();
-#endif
+        ReportError(_("Failed to unload shared library"),
```

Comment:
Won't this fail the assert you've added?

Reply:
It will, and it shouldn't. I'll fix it asap.